Terminal.Gui/Core/EscSeqUtils/EscSeqReq.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Terminal.Gui {
+	/// <summary>
+	/// Represents a state of escape sequence request.
+	/// </summary>
+	/// <remarks>
+	/// This is needed because there are some escape sequence requests responses that are equal
+	/// with some normal escape sequences and thus, will be only considered the responses to the
+	/// requests that were registered with this object.
+	/// </remarks>
+	public class EscSeqReqStat {
```

Comment:
What is a "Stat"? The name of this class is confusing to me.

Reply:
> Represents the state of an ANSI escape sequence request. Used to differntiate between request responses.

Is this accurate (and complete)? I don't understand what "that were registered with this object" means? What is "this object"? (This is a *class* not an object).

Reply:
It's the status of a request ansi escape sequence. See this:

```cs
			case "CSI":
				if (!string.IsNullOrEmpty (code) && code == "<") {
					GetMouse (cki, out buttonState, out pos, continuousButtonPressedHandler);
					isKeyMouse = true;
					return;
				} else if (escSeqReqProc != null && escSeqReqProc.Requested (terminating)) {
					isReq = true;
					escSeqReqProc.Remove (terminating);
					return;
				}

```

`escSeqReqProc` will check if there is a request escape sequence, sets the `isReq` to true if found and remove the requesting from the object. Maybe isn't perfect but was the way to differentiate from a keystroke or mouse event.

Reply:
Please fix the typo in that name. "Status" vs "Stat".

Reply:
Yesterday I already renamed from `EscSeqReqStat` to `EscSequenceRequestState`. Now do you mean to change from `EscSeqReqStat` to `EscSeqReqStatus` or from `EscSequenceRequestState` to `EscSequenceRequestStatus`?

Reply:
Name it with the correct term. It's not clear to me if it is "State" or "Status". But "Stat" is clearly incorrect.

Reply:
Done. Renamed to `EscSeqReqStatus`.